swim/node_bootstrap_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package swim
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type BootstrapTestSuite struct {
+	suite.Suite
+	tnode *testNode
+	node  *Node
+	peers []*testNode
+}
+
+func (s *BootstrapTestSuite) SetupTest() {
+	s.tnode = newChannelNode(s.T())
+	s.node = s.tnode.node
+}
+
+func (s *BootstrapTestSuite) TearDownTest() {
+	destroyNodes(append(s.peers, s.tnode)...)
+}
+
+func (s *BootstrapTestSuite) TestBootstrapOk() {
+	s.peers = genChannelNodes(s.T(), 5)
+	bootstrapNodes(s.T(), append(s.peers, s.tnode)...)
+	// Reachable members should be s.node + s.peers
+	s.Equal(6, s.node.CountReachableMembers())
+}
+
+func (s *BootstrapTestSuite) TestBootstrapTimesOut() {
+	_, err := s.node.Bootstrap(&BootstrapOptions{
+		Hosts:           fakeHostPorts(1, 1, 1, 0),
+		MaxJoinDuration: time.Millisecond,
+	})
+
+	s.Error(err, "expected bootstrap to exceed join duration")
+}
+
+func (s *BootstrapTestSuite) TestBootstrapJoinsTimeOut() {
+	_, err := s.node.Bootstrap(&BootstrapOptions{
+		Hosts:           append(fakeHostPorts(2, 2, 1, 5), s.node.Address()),
+		MaxJoinDuration: time.Millisecond,
+		JoinTimeout:     time.Millisecond / 2,
+	})
+
+	s.Error(err, "expected bootstrap to exceed join duration")
+}
+
+func (s *BootstrapTestSuite) TestBootstrapDestroy() {
+	var lerr lError
+
+	go func() {
+		_, err := s.node.Bootstrap(&BootstrapOptions{
+			Hosts:       fakeHostPorts(1, 1, 1, 10),
+			JoinTimeout: time.Millisecond,
+		})
+		lerr.Set(err)
+	}()
+
+	time.Sleep(2 * time.Millisecond)
```

Comment:
Not sure. This is a renamed file so I haven't seen this code before. It might be to induce a context switch, but that's just a guess; I don't know if time.Sleep() does this or not.